src/etc/inc/interfaces.inc

```diff
@@ -3867,7 +3867,15 @@ function interface_dhcpv6_configure($interface = "wan", $wancfg) {
 
 	$wanif = get_real_interface($interface, "inet6");
 	$dhcp6cconf = "";
-
+	
+	if ($wancfg['dhcp6-duid'] != '') {
+	// Write the DUID file
+		$dhcp6duid = $wancfg['dhcp6-duid'];
```

Comment:
There is no need to set $dhcp6duid here, you can use $wancfg['dhcp6-duid'] direct on function call